projects/01-rubygems-cli/solution/lib/ruby_gems_api.rb

```diff
@@ -0,0 +1,33 @@
+require "faraday"
+require "faraday/net_http"
+require "json"
+require "./errors/gem_not_found"
+require "./errors/bad_request"
+
+class RubyGemsApi
+  @connection = Faraday.new("https://rubygems.org") do |faraday|
+    faraday.adapter :net_http
+  end
+
+  class << self
+    def gem_info(gem_name)
```

Comment:
This method not only extracts the `gem_info`, it also validates the input (and does it extensively), so it does more than one thing. 

Can you think of a solution to fix this? 😛

Reply:
My idea is to create a private method "validate_gem_name" in which we will do the validation and call it in "gem_info". Is this approach right?

Reply:
f5c01971b89c283c4863ca608b6b8fac20e35405 This issue was resolved in branch week2.